File: app/schemas/analytics.py
"""Faz 3 — kullanım analitiği request/response şemaları."""

from __future__ import annotations

from pydantic import BaseModel, Field


class PageViewCreate(BaseModel):
    route: str = Field(..., min_length=1, max_length=255)


class RouteCount(BaseModel):
    route: str
    count: int


class PageViewStats(BaseModel):
    period_days: int
    total_views: int
    top_routes: list[RouteCount]
    bottom_routes: list[RouteCount]

File: frontend/src/api/analytics.ts
import { recordPageViewApiV1AnalyticsPageViewPost } from "../generated/api/analytics/analytics";
import axiosInstance from "../services/api/axios-instance";

/**
 * Faz 3 — sayfa görüntüleme kaydı. Best-effort: hata yutulur, UI'ı bozmaz.
 */
export async function recordPageView(route: string): Promise<void> {
  try {
    await recordPageViewApiV1AnalyticsPageViewPost({ route });
  } catch {
    // analitik best-effort — sessizce geç
  }
}

export interface RouteCount {
  route: string;
  count: number;
}

export interface PageViewStats {
  period_days: number;
  total_views: number;
  top_routes: RouteCount[];
  bottom_routes: RouteCount[];
}

export async function fetchPageViewStats(days = 30): Promise<PageViewStats> {
  const { data } = await axiosInstance.get<PageViewStats>(
    "/admin/analytics/page-views",
    { params: { days } },
  );
  return data;
}
